lib/AccessTokenClient.ts

```diff
@@ -0,0 +1,139 @@
+import { ObjectUtils } from '@sphereon/ssi-types';
+
+import { convertJsonToURI, post } from './functions';
+import {
+  AccessTokenRequest,
+  AccessTokenRequestOpts,
+  AccessTokenResponse,
+  AuthorizationServerOpts,
+  ErrorResponse,
+  GrantTypes,
+  IssuanceInitiationRequestPayload,
+  IssuanceInitiationWithBaseUrl,
+  IssuerTokenEndpointOpts,
+} from './types';
+
+export class AccessTokenClient {
+  // private _clientId?: string;
+  // private _authorizationServerUrl?: string;
+
+  public async acquireAccessTokenUsingIssuanceInitiation(
+    issuanceInitiation: IssuanceInitiationWithBaseUrl,
+    clientId: string,
+    opts?: AccessTokenRequestOpts
+  ): Promise<AccessTokenResponse | ErrorResponse> {
+    const { issuanceInitiationRequest } = issuanceInitiation;
+    const reqOpts = {
+      isPinRequired: issuanceInitiationRequest.user_pin_required || false,
+      issuerOpts: { issuer: issuanceInitiationRequest.issuer },
+      asOpts: opts?.asOpts ? { ...opts.asOpts } : undefined,
+    };
+    return await this.acquireAccessTokenUsingRequest(await this.createAccessTokenRequest(issuanceInitiationRequest, clientId, opts), reqOpts);
+  }
+
+  public async acquireAccessTokenUsingRequest(
+    accessTokenRequest: AccessTokenRequest,
+    opts: { isPinRequired?: boolean; asOpts?: AuthorizationServerOpts; issuerOpts?: IssuerTokenEndpointOpts }
```

Comment:
Create a new type of it

Reply:
Could you explain to my why?

Reply:
To be clear I can think of one reason. Just want to hear why you believe this needs to be done

Reply:
It makes it reusable and it will fit the screen, making the code look cleaner. If it was Java I would say that it would improve readability, in Typescript I'm not really sure.

Reply:
Please share your reason, so I can learn something from it.

Reply:
The only reason is consistency to be honest.
Readability really depends on the actual values. Why are we okay for instance with a function like this:

````typescript
randomFunction(arg1: string, arg2:number)
````

Why don't we say this needs an Interface for the arguments? Simply because that hides the arguments. Upon first glance you cannot see anymore what the actual arguments will be. It could be a very complex object, whilst in reality it is only 2 arguments.

In the original line for you comment it is debatable whether adding an interface for the opts is increasing readability. Right now you can see that there is a boolean , together with optional Authorization Server Options and Issuer Token Endpoints Options. At least you get a feel for what the arguments will be about. If we replace that with a singular Interface, you certainly cannot deduce what the options are about, without actually looking at the interface.

I roughly take as a rule for myself that if a function or argument object gets more than 3 params, to move it to an interface to improve readability.

There are always exceptions possible of course. For instance whenever you have cases where arguments or options can be re-used across different methods. Then it makes sense to immediately move them to an interface for re-usability.